docs/api/rest_api_authentication.md

```diff
@@ -0,0 +1,74 @@
+# REST API reference authentication basics
+
+This page refers to [REST API reference](https://doc.ezplatform.com/rest-api-reference), where you can find detailed information about
+REST API resources and endpoints.
+
+Using HTTPS for authenticated (REST) traffic is highly recommended.
+
+## Basic authentication
+
+For more information, see [HTTP Authentication: Basic and Digest Access Authentication.](http://tools.ietf.org/html/rfc2617)
+
+## OAuth
+
+For more information, see [OAuth 2.0 protocol for authorization.](https://oauth.net/2/)
+
+## Session-based authentication
+
+Sessions are created to re-authenticate the user only  (and perform authorization), not to hold session state in the service.
+Because of that, we regard this method as supporting AJAX-based applications even if it violates the principles of RESTful services.
+
+For more information, see [REST API authentication](general_rest_usage.md#rest-api-authentication).
+
+### Session cookie
+
+If activated, the user must log in, and the client must send the session cookie in every request, using standard Cookie header.
+The name (`sessionName`) and value (`sessionID`) of the header are defined in a `/user/sessions` POST response.
+
+Example request header: `Cookie: <SessionName> : <sessionID>`.
+
+### CSRF token
+
+A CSRF token must be sent in every request that uses unsafe methods (not GET or HEAD), when a session has been established.
+It should be sent with an `X-CSRF-Token` header.
+The token (`csrfToken`) is defined in a response during logging in through the POST `/user/sessions`.
+
+Example request headers:
+
+```
+DELETE /content/types/32 HTTP/1.1
+X-CSRF-Token: <csrfToken>
+```
+
+```
+DELETE /user/sessions/<sessionID>
+X-CSRF-Token: <csrfToken>
+```
+
+If an unsafe request is missing the CSRF token, or the token has incorrect value, an error is returned: `401 Unauthorized`.
+
+### Rich client application security concerns
+
+The purpose of CSRF protection is to prevent users from accidentally running harmful operations by being tricked into executing an http(s) request against a web applications they are logged into.
+In browsers this action will be blocked by lack of CSRF token.
+
+However, if you develop a rich client application (JavaScript, JAVA, Flash, Silverlight, iOS, Android, etc.), that is:
+
+- Registering itself as a protocol handler:
+    - Exposes unsafe methods in any way
+- Authenticates using either:
+    - Session-based authentication
+    - "Client side session" by remembering user login/password
+
+Then, you have to make sure to confirm with the user if they want to perform an unsafe operation.
```

Comment:
Could you elaborate on that? Do you mean that it's possible not to require CSRF token by client app? If so this is not entirely true. It depends on Symfony project [configuration of CSRF protection](https://symfony.com/doc/current/security/csrf.html). If [`framework.csrf_protection`](https://github.com/ezsystems/ezplatform/blob/v3.1.1/config/packages/framework.yaml#L4) is enabled (default), an app requesting method other than `GET`, `HEAD`, or `OPTIONS` is required to specify CSRF token. Otherwise it will result in a HTTP error:
```
HTTP 401 Unauthorized
The User does not have the '<method> /api/ezp/v2/<endpoint>' 'Missing or invalid CSRF token' permission
```

Reply:
For this one I will ask @andrerom for a comment, as this was part of his fix back in 2013: https://github.com/ezsystems/ezpublish-kernel/commit/28ede58c9a4b5960dacbbca7a8b0d744a7290e3d

Reply:
See the example below.

It's a warning for anyone writing possible client side apps that expose eZ Platform repository actions, and register them via protocol handlers.

**Implied here, which is maybe not so clear, the potential app is handling CSRF for you.**

So when you click the link in the example, the app will, unless warning the user, happily refresh session, get updated CSRF token with the session refresh,  and perform the delete with the token.